src/backend/utils/init/postinit.c

```diff
@@ -368,6 +368,22 @@ InitPostgres(const char *in_dbname, Oid dboid, const char *username,
 	char		dbname[NAMEDATALEN];
 
 	/*
+	 * User is not supposed to connect to hcatalog database,
+	 * because it's reserved for hcatalog feature integration
+	 */
+	if (!bootstrap)
+	{
+		if (strcmp(in_dbname, HcatalogDbName) == 0)
+		{
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_DATABASE),
+					errmsg("\"%s\" database is only for system use",
```

Comment:
Examples of current errors for incorrect connections attempts:
- psql: FATAL:  database "te" does not exist
- psql: FATAL:  database "template0" is not currently accepting connections

Error for "hcatalog":
psql -d hcatalog;
psql: FATAL:  "hcatalog" database is only for system use.

So I think current error is enough and consistent with other similar errors.